go-controller/pkg/ovn/multipolicy_test.go

```diff
@@ -630,12 +644,52 @@ var _ = ginkgo.Describe("OVN MultiNetworkPolicy Operations", func() {
 				err := app.Run([]string{app.Name})
 				gomega.Expect(err).NotTo(gomega.HaveOccurred())
 			},
-			ginkgo.Entry("on local zone for layer3 topology", ovntypes.Layer3Topology, false),
-			ginkgo.Entry("on remote zone for layer3 topology", ovntypes.Layer3Topology, true),
-			ginkgo.Entry("on local zone for layer2 topology", ovntypes.Layer2Topology, false),
-			ginkgo.Entry("on remote zone for layer2 topology", ovntypes.Layer2Topology, true),
-			ginkgo.Entry("on local zone for localnet topology", ovntypes.LocalnetTopology, false),
-			ginkgo.Entry("on remote zone for localnet topology", ovntypes.LocalnetTopology, true),
+			NewTestCase(ovntypes.Layer3Topology).Build(),
+			NewTestCase(ovntypes.Layer3Topology).WithRemote().Build(),
+			NewTestCase(ovntypes.Layer2Topology).Build(),
+			NewTestCase(ovntypes.Layer2Topology).WithRemote().Build(),
+			NewTestCase(ovntypes.LocalnetTopology).Build(),
+			NewTestCase(ovntypes.LocalnetTopology).WithRemote().WithOVNMapping("physnet:breth0,network1:ovsbr1").Build(),
+			NewTestCase(ovntypes.LocalnetTopology).WithRemote().ExpectFailure(
+				`Warning ErrorReconcilingPod failed to find bridge mapping for network: "network1", node "node1", ovn-bridge-mappings: ""`,
+			).Build(),
 		)
 	})
 })
+
+type TestCaseBuilder struct {
+	topology      string
+	remote        bool
+	expectedError string
+	mapping       string
+}
+
+func NewTestCase(topology string) *TestCaseBuilder {
```

Comment:
How about we try this one instead ? https://dev.to/kittipat1413/understanding-the-options-pattern-in-go-390c

Reply:
this is opinionated, you can choose to ignore.

Reply:
will consider for next times if its ok
thanks

Reply:
All good.